src/NumericalAlgorithms/Spectral/SwshTransformJob.hpp

```diff
@@ -0,0 +1,380 @@
+
+    if (spin < 0) {
+      // the conjugate is needed because libsharp doesn't handle negative spins.
+      for (size_t i = 0; i < post_transform_slices.size(); ++i) {
+        post_transform_slices[i].conjugate();
+      }
+    }
+
+    // The inverse transformed collocation data has just been placed in the
+    // memory blocks controlled by the `ComplexDataView`s. Finally, that data
+    // must be flushed back to the Variables.
+    for (size_t i = 0; i < post_transform_slices.size(); ++i) {
+      post_transform_slices[i].copy_back_to_source();
+    }
+  }
+
+ private:
+  size_t number_of_radial_grid_points_;
+  size_t l_max_;
+  sharp_alm_info* alm_info_;
+  const Collocation<Representation>* collocation_metadata_;
+};
+
+namespace detail {
+template <typename Job>
+struct transform_job_is_not_empty : std::true_type {};
+
+template <int Spin, ComplexRepresentation Representation>
+struct transform_job_is_not_empty<
+    TransformJob<Spin, Representation, tmpl::list<>>> : std::false_type {};
+
+template <int MinSpin, ComplexRepresentation Representation, typename TagList,
+          typename IndexSequence>
+struct make_swsh_transform_job_list_impl;
+
+template <int MinSpin, ComplexRepresentation Representation, typename TagList,
+          int... Is>
+struct make_swsh_transform_job_list_impl<MinSpin, Representation, TagList,
+                                         std::integer_sequence<int, Is...>> {
+  using type = tmpl::filter<
+      tmpl::list<TransformJob<Is + MinSpin, Representation,
+                              get_tags_with_spin<Is + MinSpin, TagList>>...>,
+      transform_job_is_not_empty<tmpl::_1>>;
+};
+}  // namespace detail
+
+/// \ingroup SpectralGroup
+/// \brief Assemble a `tmpl::list` of `TransformJob` given a list of tags
+/// `TagList that need to be transformed. The `Representation` is the
+/// `ComplexRepresentation` to use for the transformations.
+///
+/// \details Up to five `TransformJob` will be returned, corresponding to
+/// the possible spin values. Any number of transformations are aggregated into
+/// that set of `TransformJob`s.
+///
+/// \snippet Test_SwshTransformJob.cpp make_swsh_transform_job_list
+template <ComplexRepresentation Representation, typename TagList>
+using make_swsh_transform_job_list =
+    typename detail::make_swsh_transform_job_list_impl<
+        -2, Representation, TagList,
+        decltype(std::make_integer_sequence<int, 5>{})>::type;
+
+/// \ingroup SpectralGroup
+/// \brief Assemble a `tmpl::list` of `TransformJob`s given a list of
+/// `Derivative<Tag, Derivative>` that need to be computed. The
+/// `TransformJob`s constructed by this type alias correspond to the
+/// `Tag`s in the list.
+///
+/// \details This is intended as a convenience utility for the first step of a
+/// derivative routine, where one transforms the set of fields for which
+/// derivatives are required.
+///
+/// \snippet Test_SwshTransformJob.cpp make_swsh_transform_from_derivative_tags
```

Comment:
If you have a may "real world use" example, I think that would be better for the snippet, or as an additional snippet.